app/src/components/Modals/activity-modal/activity-form-modal.tsx

```diff
@@ -182,21 +157,43 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
 
   const onSubmit: SubmitHandler<Inputs> = async ({ activity }) => {
     const gasValues = extractGasesAndUnits(activity);
+
+    // extract field values
+    const values: Record<string, any> = {};
+    fields?.forEach((field) => {
+      if (field.id in activity) {
+        values[field.id] = (activity as any)[field.id];
+      }
+      if (field.units) {
+        values[`${field.id}unit`] = (activity as any)[`${field.id}unit`];
+      }
+    });
+
+    // so the issue here is that we need to have one inventoryValue for
     const requestData = {
-      activityData: {
-        co2_amount: gasValues[1].factor,
-        ch4_amount: gasValues[0].factor,
-        n2o_amount: gasValues[2].factor,
-        activity_type: activity.buildingType,
-        fuel_type: activity.fuelType,
-      },
-      metadata: {},
-      inventoryValue: {
-        inputMethodology: getInputMethodology(methodology?.id), // extract methodology name
-        gpcReferenceNumber: referenceNumber,
-        unavailableReason: "",
-        unavailableExplanation: "",
+      activityData: methodology?.id.includes("direct-measure")
+        ? {
+            co2_amount: gasValues[1].factor,
+            ch4_amount: gasValues[0].factor,
+            n2o_amount: gasValues[2].factor,
```

Comment:
Emissions factors are not direct measure gas values, see feedback on #648

Reply:
Created a PR for this